sentry_sdk/client.py

```diff
@@ -334,19 +334,24 @@ def capture_event(
         if session:
             self._update_session_from_event(session, event)
 
-        if event_opt.get("type") == "transaction":
-            # Transactions should go to the /envelope/ endpoint.
-            self.transport.capture_envelope(
-                Envelope(
-                    headers={
-                        "event_id": event_opt["event_id"],
-                        "sent_at": format_timestamp(datetime.utcnow()),
-                    },
-                    items=[
-                        Item(payload=PayloadRef(json=event_opt), type="transaction"),
-                    ],
-                )
+        attachments = hint.get("attachments")
+        needs_envelopes = event_opt.get("type") == "transaction" or attachments
+
+        if needs_envelopes:
+            # Transactions or envelopes with attachments should go to the
+            # /envelope/ endpoint.
+            envelope = Envelope(
+                headers={
+                    "event_id": event_opt["event_id"],
+                    "sent_at": format_timestamp(datetime.utcnow()),
+                },
+                items=[
+                    Item(payload=PayloadRef(json=event_opt), type="transaction"),
```

Comment:
This should only be done for transactions -- otherwise (error) events with attachments will be sent with type transaction.

Reply:
Yeah, this is wrong right now. I am going to push up a change to fix this but i'm starting to think that it's weird we have `add_event` and `add_transaction` instead of `add_event` figuring this out itself.

Reply:
I changed this now.